server/Player.js

```diff
@@ -196,8 +198,14 @@ class Player {
   }
 
   energyRegenerate() {
-    if (!this.isDead && this.energy < ENERGY_MAX) {
+    if (
+      this.energyRegenDelay >= 30 &&
```

Comment:
Please specify delay unit in the name, like `energyRegenDelayTicks`.
Also this `30` should be a constant.